geomstats/geometry/discrete_curves.py

```diff
@@ -842,3 +842,340 @@ def g_criterion(srv, srv_norms):
+        self.l2_metric = lambda n: L2Metric(
+            self.ambient_manifold, n_landmarks=n)
+        self.a = a
+        self.b = b
+
+    def cartesian_to_polar(self, curve):
+        """Compute polar coordinates of a curve from the cartesian ones.
+
+        See [KN2018]_ for details.
+
+        Parameters
+        ----------
+        curve : array-like, shape=[..., n_sampling_points, ambient_dim]
+            Discrete curve.
+
+        Returns
+        -------
+        norms : array-like, shape=[..., n_sampling_points]
+            Norms of the sampling points.
+        args : array-like, shape=[..., n_sampling_points]
+            Arguments of the sampling points.
+        """
+        n_sampling_points = curve.shape[-2]
+        inner_prod = self.ambient_metric.inner_product
+
+        norms = self.ambient_metric.norm(curve)
+        arg_0 = math.atan2(curve[0, 1], curve[0, 0])
+        args = [arg_0]
+
+        for i in range(1, n_sampling_points):
+            point, last_point = curve[i], curve[i - 1]
+            prod = inner_prod(point, last_point)
+            cosine = prod / (norms[i] * norms[i - 1])
+            angle = gs.arccos(gs.clip(cosine, -1, 1))
+            det = gs.linalg.det(gs.array([last_point, point]))
+            orientation = gs.sign(det)
+            arg = args[-1] + orientation * angle
+            args.append(arg)
+
+        args = gs.array(args)
+
+        return norms, args
+
+    def polar_to_cartesian(self, norms, args):
+        """Compute the cartesian coordinates of a curve from polar ones.
+
+        Parameters
+        ----------
+        norms : array-like, shape=[..., n_sampling_points]
+            Norms of sampling points.
+        args : array-like, shape=[..., n_sampling_points]
+            Arguments of sampling points.
+
+        Returns
+        -------
+        curve : array-like, shape=[..., n_sampling_points, ambient_dim]
+            Discrete curve.
+        """
+        curve_x = gs.cos(args)
+        curve_y = gs.sin(args)
+        unit_curve = gs.vstack((curve_x, curve_y)).T
+        curve = norms[:, None] * unit_curve
+
+        return curve
+
+    def f_transform(self, curve):
+        """Compute the F_transform of a curve.
+
+        See [KN2018]_ for details.
+
+        Parameters
+        ----------
+        curve : array-like, shape=[..., n_sampling_points, ambient_dim]
+            Discrete curve.
+
+        Returns
+        -------
+        f : array-like, shape=[..., n_sampling_points - 1, ambient_dim]
+            F_transform of the curve..
+        """
+        velocity = curve[1:] - curve[:-1]
+        speeds, args = self.cartesian_to_polar(velocity)
+
+        f_args = args * self.a / (2 * self.b)
+        f_norms = 2 * self.b * gs.sqrt(speeds)
+        f = self.polar_to_cartesian(f_norms, f_args)
+
+        return f
+
+    def f_transform_inverse(self, f, starting_point):
+        """Compute the inverse F_transform of a transformed curve.
+
```

Comment:
If we have an analytic form for the inverse, could you add it in :math: in the docstring too? thank you! 🙌